smriprep/interfaces/fastsurfer.py

```diff
@@ -0,0 +1,673 @@
+        desc="White matter parcellation statistics file",
+    )
+    aparc_stats = OutputMultiPath(
+        File(exists=True),
+        loc="stats",
+        altkey="aparc.DKTatlas.mapped",
+        desc="Aparc parcellation statistics files",
+    )
+    curv_stats = OutputMultiPath(
+        File(exists=True),
+        loc="stats",
+        altkey="curv",
+        desc="Curvature statistics files"
+    )
+    w_g_stats = OutputMultiPath(
+        File(exists=True),
+        loc="stats",
+        altkey="w*g.pct",
+        desc="White minus gray statistics files"
+    )
+    aseg_presurf_stats = OutputMultiPath(
+        File(exists=True),
+        loc="stats",
+        altkey="aseg.presurf.hypos",
+        desc="Automated segmentation pre-surface recon statistics files"
+    )
+    sd = Directory(
+        exists=True,
+        argstr="--sd %s",
+        desc="Subjects directory"
+    )
+    sid = traits.String(
+        exists=True,
+        argstr="--sid %s",
+        desc="Subject ID"
+    )
+
+
+class FastSurferSource(IOBase):
+    """Generates FastSurfer subject info from their directories.
+
+    """
+
+    input_spec = FastSurfSourceInputSpec
+    output_spec = FastSurfSourceOutputSpec
+    _always_run = True
+    _additional_metadata = ["loc", "altkey"]
+
+    def _get_files(self, path, key, dirval, altkey=None):
+        globsuffix = ""
+        if dirval == "mri":
+            globsuffix = ".mgz"
+            if key in ("mni_log_bak"):
+                globsuffix = ".log.bak"
+        elif dirval == "stats":
+            globsuffix = ".stats"
+        globprefix = ""
+        if dirval in ("surf", "label", "stats"):
+            globprefix = "?h."
+        if key in ("aseg_stats", "aseg_presurf_stats", "wmparc_stats"):
+            globprefix = ""
+        elif key == "ribbon":
+            globprefix = "*"
+        elif key == "aparc_ctab":
+            globprefix = ""
+            globsuffix = ""
+        elif key in ("nofix", "defects", "preaparc"):
+            globsuffix = "*"
+        keys = ensure_list(altkey) if altkey else [key]
+        globfmt = os.path.join(path, dirval, "".join((globprefix, "{}", globsuffix)))
+        return [
+            os.path.abspath(f) for key in keys for f in glob.glob(globfmt.format(key))
+        ]
+
+    def _list_outputs(self):
+        subject_path = os.path.join(self.inputs.sd, self.inputs.sid)
+        output_traits = self._outputs()
+        outputs = output_traits.get()
+        for k in list(outputs.keys()):
+            val = self._get_files(
+                subject_path,
+                k,
+                output_traits.traits()[k].loc,
+                output_traits.traits()[k].altkey,
+            )
+            if val:
+                outputs[k] = simplify_list(val)
+        return outputs
+
+
+class FastSCommand(CommandLine):
+    """
+    Wraps FastSurfer command for segmentation and surface processing
+
+    """
+
+    input_spec = FastSurfSourceInputSpec
```

Comment:
I think this supposed to be:

```suggestion
    input_spec = FastSInputSpec
```